ddtrace/debugging/_expressions.py

```diff
@@ -132,17 +149,14 @@ def _compile_arg_predicate(ast):
     if _type in {"any", "all"}:
         a, b = args
         f = __builtins__[_type]  # type: ignore[index]
-        ca, fb = _compile_predicate(a), _make_lambda(b)
+        ca, cb = _compile_predicate(a), _compile_lambda_with_locals(b)
+
         if ca is None:
             raise ValueError("Invalid argument: %r" % a)
-        return (
-            [Instr("LOAD_CONST", lambda i, c: f(c(_) for _ in i))]
-            + ca
-            + [
-                Instr("LOAD_CONST", fb),
-                Instr("CALL_FUNCTION", 2),
-            ]
-        )
+        if cb is None:
+            raise ValueError("Invalid lmabda: %r" % b)
+
+        return [Instr("LOAD_CONST", lambda i, c: f(c(_) for _ in i))] + ca + cb + [Instr("CALL_FUNCTION", 2)]
```

Comment:
I think we could probably simplify this slightly by doing something like
```suggestion
        return [Instr("LOAD_CONST", lambda i, c, _locals: f(c(_, _locals) for _ in i))] + ca + cb + [Instr("CALL_FUNCTION", 2)]
```
and the pushing a third argument to the stack, before `CALL_FUNCTION` with
```
Instr("LOAD_FAST", "_locals")
```
Then I think we shouldn't need the new helpers. WDYT?